src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4423,12 +4431,12 @@ private ParameterSyntax ParseParameter()
                     // .Identifier has the kind ArgListKeyword.
                     type = null;
                     name = this.EatToken(SyntaxKind.ArgListKeyword);
+                    if (this.CurrentToken.Kind == SyntaxKind.EqualsToken)
+                    {
+                        equals = this.EatToken(SyntaxKind.EqualsToken);
+                    }
                 }
 
-                if (this.CurrentToken.Kind == SyntaxKind.EqualsToken)
-                {
-                    equals = this.EatToken(SyntaxKind.EqualsToken);
-                }
                 if (exclamationExclamation != null)
                 {
                     exclamationExclamation = (exclamationExclamation.Kind != SyntaxKind.ExclamationExclamationToken)
```

Comment:
i don't see hwo this case ever happens.  eerything that sets exclamationExclamation has already made an ExclamationExclamationToken.  Indeed, i think we can have a contract call for exactly that.